### PR TITLE
BSTOR-33: Checkout transitions

### PR DESCRIPTION
**Pull request by Elrhino**

### CODE REVIEW

File: client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/SidePanelView.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.arcbees.beeshop.client.application.widget.sidepanel;
+
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.HTMLPanel;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+import com.gwtplatform.mvp.client.ViewWithUiHandlers;
+
+public class SidePanelView extends ViewWithUiHandlers<SidePanelUiHandlers>
```
Comment (jclariviere) [inline client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/SidePanelView.java]: Extend `ViewImpl` instead, you don't need uiHandlers.

File: client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/SidePanelView.ui.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<ui:UiBinder xmlns:ui='urn:ui:com.google.gwt.uibinder'
+        xmlns:g='urn:import:com.google.gwt.user.client.ui'>
+
+    <ui:with field="res" type="com.arcbees.beeshop.client.resources.AppResources"/>
+
+    <g:HTMLPanel styleName="{res.style.rightPanel}">
+        <g:HTMLPanel ui:field="main"/>
```
Comment (jclariviere) [inline client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/SidePanelView.ui.xml]: You can probably remove this HTMLPanel and set the ui:field on the parent.

File: client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagModule.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package com.arcbees.beeshop.client.application.widget.sidepanel.cart;
+
+import com.gwtplatform.mvp.client.gin.AbstractPresenterModule;
+
+public class ShoppingBagModule extends AbstractPresenterModule {
+    @Override
+    protected void configure() {
+        bind(ShoppingBagItemPresenter.MyView.class).to(ShoppingBagItemView.class);
```
Comment (jclariviere) [inline client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagModule.java]: bindPresenterWidget

File: client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagView.java [hunk 4]
```diff
@@ -41,10 +46,25 @@
...
+            }
+        });
     }
 
     @Override
     public void updateItemNumber(int number) {
         $(numberOfItems).text(String.valueOf(number));
     }
+
+    public void onCheckoutButtonClick(ClickEvent event) {
```
Comment (jclariviere) [inline client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagView.java]: rm

File: client/src/test/java/com/arcbees/beeshop/client/application/widget/SidePanelPresenterTest.java [hunk 5]
```diff
@@ -15,7 +15,7 @@
 import com.gwtplatform.mvp.client.AutobindDisable;
 
 @RunWith(JukitoRunner.class)
-public class PricePresenterTest {
+public class SidePanelPresenterTest {
```
Comment (jclariviere) [inline client/src/test/java/com/arcbees/beeshop/client/application/widget/SidePanelPresenterTest.java]: Rename back :)

**Comment by jclariviere** [comment]
LGTM

File: client/src/main/java/com/arcbees/beeshop/client/application/ApplicationPresenter.java [hunk 6]
```diff
@@ -52,9 +52,9 @@
     }
 
     public static final NestedSlot SLOT_MAIN = new NestedSlot();
-    public static final SingleSlot SLOT_CART_WIDGET = new SingleSlot();
+    public static final SingleSlot SLOT_SIDE_PANEL = new SingleSlot();
 
-    private ShoppingBagPresenter shoppingBagPresenter;
+    private final SidePanelPresenter sidePanelPresenter;
     private CurrentShoppingBag currentShoppingBag;
```
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/ApplicationPresenter.java]: nl

File: client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagView.java [hunk 7]
```diff
@@ -41,6 +45,17 @@
         initWidget(binder.createAndBindUi(this));
 
         bindSlot(ShoppingBagPresenter.SLOT_BAG_ITEM, itemsContainer);
+
+        bind();
+    }
+
+    private void bind() {
```
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagView.java]: Why don't you guys use a button and `@UiHandler` ?

File: client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagView.java [hunk 8]
```diff
@@ -41,6 +45,17 @@
...
+        $(checkoutButton).click(new Function() {
+            @Override
+            public void f() {
+                getUiHandlers().checkout();
```
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagView.java]: Should use `onCheckout()` standard.

File: client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagPresenter.java [hunk 9]
```diff
@@ -61,6 +62,11 @@ public void onShoppingBagChanged(ShoppingBagChangedEvent event) {
     }
 
     @Override
+    public void checkout() {
+        CheckoutContinueEvent.fire(this);
+    }
+
+    @Override
     protected void onBind() {
         addRegisteredHandler(ShoppingBagChangedEvent.TYPE, this);
```
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagPresenter.java]: Could it be a visibleHandler ?

File: client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/checkout/AddressView.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        initWidget(binder.createAndBindUi(this));
+
+        bind();
+    }
+
+    private void bind() {
+        $(continueButton).click(new Function() {
+            @Override
+            public void f() {
+                getUiHandlers().onContinueClicked();
```
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/checkout/AddressView.java:84]: g:button with uiHandler?

File: client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/checkout/OrderView.java [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+        initWidget(binder.createAndBindUi(this));
+
+        bind();
+    }
+
+    private void bind() {
+        $(continueButton).click(new Function() {
+            @Override
+            public void f() {
+                getUiHandlers().onContinueClicked();
```
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/checkout/OrderView.java:49]: same here

File: client/src/main/java/com/arcbees/beeshop/client/events/CheckoutContinueEvent.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+package com.arcbees.beeshop.client.events;
+
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HasHandlers;
+
+public class CheckoutContinueEvent extends GwtEvent<CheckoutContinueEventHandler> {
+    public static final Type<CheckoutContinueEventHandler> TYPE = new Type<>();
+
+    public CheckoutContinueEvent() {
```
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/events/CheckoutContinueEvent.java]: rm if empty

File: client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagPresenter.java (hunk 9 above)
Comment (jclariviere) [inline client/src/main/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/ShoppingBagPresenter.java]: Nope, it's the only thing that adds items to the shopping cart, and the cart is not always visible.

**Pull request merged**